vetiver/vetiver_model.py

```diff
@@ -76,31 +75,44 @@ def __init__(
         self.model_name = model_name
         self.description = description if description else translator.describe()
         self.versioned = versioned
-        self.metadata = (
-            metadata
-            if metadata
-            else translator.create_meta(metadata, required_pkgs=["vetiver"])
```

Comment:
Here--like in R pins--it seems that if metadata is passed, it simply gets added to the `user` metadata.

However, in the new code it seems like it both gets added to the `user` metadata and potentially is used to override other model information if possible (e.g. version, url, required_pkgs).

Is the new behavior part of the piece "fixing the bug where user data is duplicated"?

Let's spend some of our pairing time going over this part (and the places metadata is created / stored / unpacked)

Reply:
Yep, that is the main reason! When using the classmethod from_pin, sometimes the metadata is already "made" (since it's piecing back together a previously created vetivermodel)